meshnets/process.py

```diff
@@ -75,9 +75,30 @@ def main(_):
             folder, FLAGS.interpolated_pressure_field_name)
         interpolated_mesh.save(interpolated_mesh_path)
 
-        graph_data = data_processing.mesh_file_to_graph_data(
-            interpolated_mesh_path, WIND_VECTOR, load_pressure=True)
+        edge_index, edge_attr = data_processing.make_edge_index_and_features(
+            interpolated_mesh_path)
+
+        wind_vector_path = os.path.join(folder, FLAGS.wind_vector_name)
+        if os.path.exists(wind_vector_path):
+            with open(wind_vector_path, encoding='utf-8') as f:
+                wind_vector = json.load(f)
+        else:
+            wind_vector = FLAGS.default_wind_vector
+        node_features = data_processing.make_node_features(
+            interpolated_mesh_path, wind_vector)
 
+        target = data_processing.make_target(interpolated_mesh_path)
+
+        # Save as numpy arrays.
+        np.save(os.path.join(folder, 'edge_index.npy'), edge_index)
+        np.save(os.path.join(folder, 'edge_attr.npy'), edge_attr)
+        np.save(os.path.join(folder, 'node_features.npy'), node_features)
```

Comment:
For node_features, I think we might be coupling what is a dataset generation step with what is an ML modelling choice. We want to store in the dataset what was the wind speed, ideally in a compact way, without making the assumption that this will be fed to the model via passing the same vector to every node in the graph. There are other choices of inputing that info the model that are different, so the dataset itself could try to be agnostic to those later choices.

Btw: is there a reason that for edges we use "attributes" but for nodes we use "features".

Reply:
> Btw: is there a reason that for edges we use "attributes" but for nodes we use "features".

Not really. In pytorch they use `edge_attr` for the edge attributes/features that is why I used the "attributes" for the edges. However, for node "attributes/features" they use only `x`. To prevent such a generic name I just wrote "features" because that is what is most common in literature

> For node_features, I think we might be coupling what is a dataset generation step with what is an ML modelling choice. We want to store in the dataset what was the wind speed, ideally in a compact way, without making the assumption that this will be fed to the model via passing the same vector to every node in the graph.

I don't necessarily agree here. The scripts in `data_generation` are the ones responsible by data generation. This script handles the post processing of said data. I don't think that there is anything wrong with having a `process` script that processes the data in a way that will be useful in our pipeline later on even if it makes assumption on how to solve the problem. Here we just process the raw data and package it in a way that will be easily handled by HuggingFaces datasets. Most hugging face datasets have their data already packaged in some way that makes loading them easy. Take a look for example at [this](https://huggingface.co/datasets/graphs-datasets/MUTAG) dgraph dataset.

Reply:
See my other comment: I think that the ML code should receive an HuggingFace dataset already generated (and it should even be done in a separate repo). Because that's what other ML engineers, e.g. outside Inductiva, can directly access. Basically that's the dataset interface to the world.

Therefore, the dataset should make little or no assumptions about the ML models that will consume it (some people might even use MLPs or Convnets or something else). The dataset should just contain all relevant data.

For example, if you want to take the wind speed and pass it as a node feature to every node: what seems more natural is that this would be done in one of the .map operations on the HuggingFace dataset before you pass it to the meshnet. No? But other ML architectures, e.g. concatenating the wind vector to a bottleneck layer of an encoder, would just use it directly, no need to have that in node features.

Reply:
Augusto, it is not because we find examples of HuggingFace datasets that hardcode assumptions about the models that consume them, that this is a great decision. We could also, literally, find millions of GitHub repos that contain poor quality code, or that do not follow our programming guidelines, or that are poorly documented, or that don't have unit tests, etc. 

If the data is a graph, the dataset will contain graph data. But graph data can be consumed by other architectures that are not GraphNets (MLPs could use tricks like padding the data to the maximum size, and convnets could deal with varying size graphs, and then have pooling layers to reduce to a smaller, fixed size), and it can be consumed by many different variants of GraphNets.

Therefore, having graph data in the dataset, does not mean we should hardcoded decisions about downstream use by particular models, particularly what is the trick to pass "global" information to the network (e.g. passing as node features everywhere, or just concatenating to a embedding vector in the bottleneck layer of the network). Do you see the difference? What are you going to do when you want to compare different ways of encoding that info in the network, regenerate a dataset in a different format?

Reply:
I will close this pull request, move the generation and post-processing code to a new repo and, while I iterate on that we can schedule a meeting to better discuss exactly what should be returned by the HuggingFace datasets